### PR TITLE
Pom 467 add phone national number

### PR DESCRIPTION
**Pull request by mndzielski**

### CODE REVIEW

File: src/main/java/pl/gov/coi/pomocua/ads/accomodations/AccommodationOfferDefinitionDTO.java [hunk 1]
```diff
@@ -50,6 +51,7 @@ public void applyTo(AccommodationOffer offer) {
         offer.guests = guests;
         offer.lengthOfStay = lengthOfStay;
         offer.hostLanguage = new LinkedList<>(hostLanguage);
-        offer.phoneNumber = Optional.ofNullable(phoneNumber).map(PhoneUtil::normalize).orElse(null);
+        offer.phoneNumber = PhoneUtil.getPhoneDetails(phoneNumber).map(PhoneDetails::nationalNumber).orElse(null);
+        offer.phoneCountryCode = PhoneUtil.getPhoneDetails(phoneNumber).map(PhoneDetails::countryCode).orElse(null);
```
Comment (sfc-gh-pnescior) [inline src/main/java/pl/gov/coi/pomocua/ads/accomodations/AccommodationOfferDefinitionDTO.java:55]: Maybe `PhoneUtil.getPhoneDetails(...)` could be done just once to set both of the fields?
Like:
```
PhoneUtil.getPhoneDetails(...).ifPresent(phoneDetails -> {
  offer.phoneNumber = phoneDetails.nationalNumber();
  offer.countryCode = phoneDetails.countryCode();
});
```

File: src/main/java/pl/gov/coi/pomocua/ads/transport/TransportResource.java [hunk 2]
```diff
@@ -61,7 +63,8 @@ public void delete(@PathVariable Long id) {
 
     @GetMapping("transport/{id}")
     public TransportOffer get(@PathVariable Long id) {
-        return repository.findById(id).filter(BaseOffer::isActive).orElseThrow(OfferNotFoundException::new);
+        Optional<TransportOffer> byId = repository.findById(id);
```
Comment (sfc-gh-pnescior) [inline src/main/java/pl/gov/coi/pomocua/ads/transport/TransportResource.java:66]: Is this change necessary?

File: src/main/java/pl/gov/coi/pomocua/ads/BaseOffer.java [hunk 3]
```diff
@@ -73,6 +77,16 @@ public void attachTo(User user) {
         this.userFirstName = user.firstName();
     }
 
+    @PostLoad
+    private void onLoad() {
+        Optional<PhoneDetails> phoneDetailsOpt = PhoneUtil.getPhoneDetails(phoneNumber);
+        if (phoneDetailsOpt.isPresent() && StringUtils.isEmpty(phoneCountryCode)) {
```
Comment (sfc-gh-pnescior) [inline src/main/java/pl/gov/coi/pomocua/ads/BaseOffer.java:83]: Maybe we could check if `StringUtils.isEmpty(phoneCountryCode)` before parsing phone number with `PhoneUtil`?
This parsing could be a heavy operation which takes time, so it would be great to do it only if necessary.

Also, `Optional` has a `ifPresent()` method which could make this code a little bit prettier, but that's just my preference.

Comment (mndzielski) [reply src/main/java/pl/gov/coi/pomocua/ads/BaseOffer.java:83]: good idea

**Review by sfc-gh-pnescior — APPROVED** [review]
